pkg/cli/extension/list.go

```diff
@@ -0,0 +1,148 @@
+package extension
+
+import (
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/jedib0t/go-pretty/v6/table"
+	"github.com/spf13/cobra"
+
+	"github.com/fuseml/fuseml-core/gen/extension"
+	"github.com/fuseml/fuseml-core/pkg/cli/client"
+	"github.com/fuseml/fuseml-core/pkg/cli/common"
+)
+
+type extensionListOptions struct {
+	client.Clients
+	global *common.GlobalOptions
+	format *common.FormattingOptions
+	query  extension.ExtensionQuery
+}
+
+func newExtensionListOptions(o *common.GlobalOptions) (res *extensionListOptions) {
+	res = &extensionListOptions{global: o}
+	res.format = common.NewFormattingOptions(
+		[]string{"ID", "Product", "Version", "Zone", "Services", "Endpoints", "Credentials"},
+		[]table.SortBy{{Name: "ID", Mode: table.Asc}},
+		common.OutputFormatters{"Services": formatServices, "Endpoints": formatEndpoints, "Credentials": formatCredentials},
+	)
+
+	return
+}
+
+func derefString(s *string, defaultValue ...string) string {
```

Comment:
couldn't you use `derefString` that you've already defined somewhere else?

Reply:
took even more changes, but I moved all ref/deref functions to the common util package